R/HintikkaXOData.R

```diff
@@ -35,42 +33,61 @@
 #' \emph{International Journal of Environmental Research and Public Health}
 #' 18(8):4049 \url{https://doi.org/10.3390/ijerph18084049}
 #'
+#' @importFrom SummarizedExperiment rowData
+#' @importFrom SummarizedExperiment rowData<-
+#' 
 #' @name HintikkaXOData
-#' @importFrom SingleCellExperiment altExp
-#' @importFrom SingleCellExperiment altExp<-
 #' @export
 #'
 #' @examples
-#' # tse <- HintikkaXOData()
+#' # Retrieve the MAE data
+#' mae <- HintikkaXOData()
+#'
+#' # List the experiments in this MultiAssayExperiment object
+#' print(experiments(mae))
+#'
+#' # colData for this MAE data object
+#' colData(mae)
+#'
+#' # metabolite assay data
+#' # assays(mae[["metabolites"]])$assay
+#'
+#' # biomarker assay data
+#' # assays(mae[["biomarkers"]])$assay
+# '
+#' # microbiome assay counts
+#' # assays(mae[["microbiome"]])$counts
 #'
+#' # microbiome rowData
+#' rowData(mae[["microbiome"]])
+#' 
 HintikkaXOData <- function() {
 
-    # Circumvent warnings
-    meta_cecum <- otu_cecum <- tax <- NULL
-
-    dataset <- "3.14/hintikka-xo"
-    se <- .create_se(dataset,
-                    assays = c("counts"),
-                    has.rowdata = TRUE,
-                    has.coldata = TRUE)
-
-    tse <- SummarizedExperiment(assays = list(counts = otu_cecum),
-                        colData = meta_cecum,
-                        rowData = tax)
+    mae <- .create_mae("3.14/hintikka-xo",
+    
+        types = list(microbiome  = list("TSE" = c("counts")),
+                    metabolites = list("SE"  = NULL),
```

Comment:
These values cannot be `NULL`. They have to reflect the assay name, which should also be used in the file name.